### PR TITLE
refactor(sg): added anti-user-gateway

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: src/lib/util/AntiUserGateway.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		return this.create();
+	}
+
+	public get(): Settings | null {
+		return null;
+	}
+
+	public create(): Settings {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
```
Comment (favna) [inline src/lib/util/AntiUserGateway.ts]: Add a short description as to why there is a ts-ignore then eslint won't complain either.

File: src/lib/util/AntiUserGateway.ts [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		return this.gateway.schema;
+	}
+
+	public get size() {
+		return 0;
+	}
+
+	public get base(): Settings | null {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
```
Comment (favna) [inline src/lib/util/AntiUserGateway.ts]: Add a short description as to why there is a ts-ignore then eslint won't complain either.

File: src/lib/util/AntiUserGateway.ts [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+/* eslint-disable @typescript-eslint/class-literal-property-style */
+
+// Step 1: Remove UserStore#fetch
+Reflect.deleteProperty(KlasaUserStore.prototype, 'fetch');
+
+// Step 2: Create UserSettings, which is an empty Settings-like class, implements:
+//         Settings, SettingsFolder, and Map<string, unknown>.
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```
Comment (favna) [inline src/lib/util/AntiUserGateway.ts]: Add a short description as to why there is a ts-ignore then eslint won't complain either.

**Review by favna — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

**Comment by kyranet** [comment]
This pull request needs to be reverted with our latest changes to our fork.

**Pull request merged**